oidc-controller/api/routers/oidc.py

```diff
@@ -135,22 +163,52 @@ async def get_authorize(request: Request, db: Database = Depends(get_db)):
             },
         )
 
-    pres_exch_dict = response.model_dump()
-
-    # Prepare the presentation request
     use_public_did = not settings.USE_OOB_LOCAL_DID_SERVICE
-    oob_invite_response = client.oob_create_invitation(pres_exch_dict, use_public_did)
-    msg_contents = oob_invite_response.invitation
+
+    if settings.USE_CONNECTION_BASED_VERIFICATION:
+        # NEW: Connection-based verification flow
+        oob_invite_response = client.create_connection_invitation(
+            multi_use=False,
+            presentation_exchange=None,  # No attachment - establish connection first
+            use_public_did=use_public_did,
+            auto_accept=True,  # Auto-accept connections to avoid manual acceptance
+        )
+        msg_contents = oob_invite_response.invitation
+
+        # We'll create the presentation request after connection is established
+        pres_exch_dict = None
+        # Use invitation message ID as temporary unique identifier
+        pres_ex_id = f"pending-{oob_invite_response.invi_msg_id}"
+    else:
+        assert False
+        # EXISTING: Out-of-band verification flow
+        response = client.create_presentation_request(proof_request)
+        pres_exch_dict = response.model_dump()
+        pres_ex_id = response.pres_ex_id
+
+        oob_invite_response = client.oob_create_invitation(
+            pres_exch_dict, use_public_did
+        )
+        msg_contents = oob_invite_response.invitation
```

Comment:
We get a 500 if this flag is set to false, are we just preserving the existing code for a bit in case we need to go back to it? Otherwise If setting the flag false will break calls then is there a point of the flag?

Unless I'm missing (I just tried setting it false and starting up and logging in), we should probably have 
- flag: False -> actually use OOB invite proof, flag: True -> use new
- no flag and deprecate pathway entirely

Seems like the `USE_CONNECTION_BASED_VERIFICATION` flag in this setup is just a "break the API" flag

Reply:
Ah ok, so we'll be still making the old OOB format available, flag usage makes sense then if we remove the assert